internal/api/v20240610preview/hcpopenshiftclusters_methods.go

```diff
@@ -315,11 +317,48 @@ func (c *HcpOpenShiftClusterResource) Normalize(out *api.HCPOpenShiftCluster) {
 // validateStaticComplex performs more complex, multi-field validations than
 // are possible with struct tag validation. The returned CloudErrorBody slice
 // contains structured but user-friendly details for all discovered errors.
-func validateStaticComplex(normalized *api.HCPOpenShiftCluster, updating bool) []arm.CloudErrorBody {
+func validateStaticComplex(normalized *api.HCPOpenShiftCluster) []arm.CloudErrorBody {
 	var errorDetails []arm.CloudErrorBody
-
-	// FIXME Perform user-assigned managed identity validation checks.
-
+	//Generate a slice of Resource IDs of ControlplaneOperators MI only , disregard the DataPlaneOperators.
+	var cpmiResourceIDs []string
+	if normalized.Identity.UserAssignedIdentities != nil {
+		for _, operatorResourceID := range normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ControlPlaneOperators {
+			cpmiResourceIDs = append(cpmiResourceIDs, operatorResourceID)
+		}
+		smiResourceID := normalized.Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities.ServiceManagedIdentity
+		miCount := len(normalized.Identity.UserAssignedIdentities)
+		for resourceID, _ := range normalized.Identity.UserAssignedIdentities {
+
+			if slices.Contains(cpmiResourceIDs, resourceID) || resourceID == smiResourceID {
+				//Duplicated Identity Validation
+				if slices.Contains(cpmiResourceIDs, resourceID) && resourceID == smiResourceID {
+					errorDetails = append(errorDetails, arm.CloudErrorBody{
+						Message: fmt.Sprintf(
+							"%s can not be used in both controlPlaneOperators and serviceManagedIdentity",
+							resourceID),
+						Target: "Properties.Spec.Platform.OperatorsAuthentication.UserAssignedIdentities",
```

Comment:
Sorry to be super nit-picky.  The `Target` field is meant to show the JSON path to the end-user, so it should be cased like the `json:` struct tags (e.g. see my suggested change for this line).

Also, just to acknowledge, hard-coding these fields is not ideal.  It would be better to generate the appropriate `Target` values (with correct casing) through some kind of [reflect](https://pkg.go.dev/reflect)-based struct field lookup routine.  But hard-coding will do for now.

```suggestion
						Target: "properties.spec.platform.operatorsAuthentication.userAssignedIdentities",
```